File: Easy/0058 - lengthoflastword/lengthoflastword.cpp
#include <string>

class Solution {
    public:
        int lengthOfLastWord(std::string s)
        {
            int     k = 0;
            bool    flag = 0;

            for (int i = s.size() - 1; i >= 0; i--)
            {
                if (s[i] == ' ' && !flag)
                    continue;
                if (s[i] != ' ')
                {
                    if (!flag)
                        flag = 1;
                    k++;
                }
                else
                    break;
            }

            return k;
        }
};

File: Easy/0058 - lengthoflastword/lengthoflastword.py
class Solution:
    def lengthOfLastWord(self, s: str) -> int:
        k = 0
        flag = 0

        for i in range(len(s) - 1, -1, -1) :
            if (s[i] == ' ' and flag == 0) :
                continue
            if (s[i] != ' ') :
                if (flag == 0) :
                    flag = 1
                k += 1
            else :
                break

        return k